File: R/dir_SWC2ABA_plots/SWC2ABA_plots.R
library("pracma")

source("select_file_csv.R")
source("plot_morphology_matrix.R")

csvFileName <- select_file_csv()
if (csvFileName == "!") {
  return()
}

idx <- strfind(csvFileName, "ARA")
if (length(idx) > 0) {
  plotLabel <- "ARA_Old"
} else {
  plotLabel <- "Atlas_V3"
}

cells <- read.csv(file=csvFileName, header=TRUE, sep=",")
print(cells)

nRows <- nrow(cells)
nColumns <- ncol(cells)

neuronLabels <- cells[1:nRows,1]
print(neuronLabels)

parcelLabels <- colnames(cells)
parcelLabels <- parcelLabels[2:nColumns]
print(parcelLabels)

morphologyMatrix <- cells[1:nRows,2:nColumns]
print(morphologyMatrix)

plot_morphology_matrix(morphologyMatrix, neuronLabels, parcelLabels, plotLabel)

File: MATLAB/dir_SWC2ABA_plots/SWC2ABA_plots.m
function SWC2ABA_plots()

    clear all;    

    addpath(genpath('lib'));
    
    csvFileName = select_file_csv();
    if strcmp(csvFileName, '!')
        return;
    end
    
    if strfind(csvFileName, 'ARA')
        plotLabel = 'ARA\_Old';
    else
        plotLabel = 'Atlas\_V3';
    end
    
    [cells, isFileLoaded] = load_csv_file(csvFileName);
    
    if isFileLoaded
       
        [nRows, nColumns] = size(cells);

        nSkipRows = 1;        
        nSkipColumns = 1;
        
        neuronColumnNo = 1;
        parcelRowNo = 1;
    
        nNeurons  = nRows - nSkipRows;
        nParcels = nColumns - nSkipColumns;
        
        neuronLabels = cells(nSkipRows+[1:nNeurons], neuronColumnNo);
        parcelLabels = cells(parcelRowNo, nSkipColumns+[1:nParcels]);
        
        morphologyMatrix = cell2mat(cells(nSkipRows+[1:nNeurons], nSkipColumns+[1:nParcels]));

        plot_morphology_matrix(morphologyMatrix, neuronLabels, parcelLabels, plotLabel);
        
    end

end % SWC2ABA_plots()